prompting/llms/model_manager.py

```diff
@@ -144,9 +147,9 @@ def generate(
         if not model:
             model = ModelZoo.get_random(max_ram=self.total_ram)
 
-        model: vllm.LLM = self.get_model(model)
+        model: ReproducibleVLLM = self.get_model(model)
         responses = model.generate(prompts=prompts, sampling_params=sampling_params)
-        return [r.outputs[0].text.strip() for r in responses]
+        return [responses]
```

Comment:
I would suggest to make an Adapter pattern, or the same interface for vLLM and your custom class, so that we can easily swap between them in future. But you can ignore this comment for this PR

Reply:
The reason why I didn't bother making this nice is because we'll have to replace vllm anyways so the structures will have to change either way